app/gameengine.py

```diff
@@ -4780,6 +4850,54 @@ def validate_main_step_place_holomem(self, player_id:str, action_data:dict):
 
         return True
 
+    def validate_main_step_attached_action(self, player_id: str, action_data: dict) -> bool:
+        if not self.validate_decision_base(player_id, action_data, DecisionType.DecisionMainStep, GameAction.MainStepAttachedActionFields):
+            return False
+
+        player = self.get_player(player_id)
+
+        # Validate that the support card exists
+        support_id = action_data["support_id"]
+        support_card = player.find_attachment(support_id)
+        if not support_card:
+            self.send_event(self.make_error_event(player_id, "invalid_action", "Support card not found."))
+            return False
+
+        # Validate that the action is part of the current decision's available actions
+        effect_id = action_data["effect_id"]
+        action_found = False
+        for action in self.current_decision["available_actions"]:
+            if action["action_type"] == GameAction.MainStepAttachedAction \
+                and action["effect_id"] == effect_id and action["support_id"] == support_id:
+                action_found = True
+        if not action_found:
+            self.send_event(self.make_error_event(player_id, "invalid_action", "Invalid action."))
+            return False
+
+        return True
+
+    def handle_main_step_attached_action(self, player_id: str, action_data: dict):
+        if not self.validate_main_step_attached_action(player_id, action_data):
+            return False
+
+        continuation = self.clear_decision()
+
+        player = self.get_player(player_id)
+        support_id = action_data["support_id"]
+        effect_id = action_data["effect_id"]
+
+        action_effects = player.get_support_attached_action_effects(support_id, effect_id)
```

Comment:
and here you'll just GetAvailableSpecialActions and grab the right action form there and do it.